Snowflake.Data/Core/SFDataConverter.cs

```diff
@@ -19,95 +19,100 @@ public enum SFDataType
 
     class SFDataConverter
     {
-        static private ILog logger = LogManager.GetLogger<SFDataConverter>();
+        private static readonly ILog Logger = LogManager.GetLogger<SFDataConverter>();
 
-        static private DateTime unixEpoch = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
-
-        static internal Object convertToCSharpVal(string srcVal, SFDataType srcType, Type destType)
+        private static readonly DateTime UnixEpoch = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+        
+        
+        internal static object ConvertToCSharpVal(string srcVal, SFDataType srcType, Type destType)
         {
-            logger.TraceFormat("src value: {0}, srcType: {1}, destType: {2}", srcVal, srcType, destType);
-            if (destType == typeof(Int16) 
-                || destType == typeof(Int32)
-                || destType == typeof(Int64)
+            Logger.TraceFormat("src value: {0}, srcType: {1}, destType: {2}", srcVal, srcType, destType);
+            
+            if (srcVal == null)
+                return DBNull.Value;

```

Comment:
It looks like this line is the actual fix, right? I will add testcase. Thanks for fixing!